src/proofnetwork/restclient.cpp

```diff
@@ -655,6 +739,16 @@ QPair<QString, QString> RestClientPrivate::parseHost(const QString &host)
     return {newHost, postfix};
 }
 
+void RestClientPrivate::sendMailAboutSlowNetwork(QNetworkReply *reply, long timeout)
+{
+    auto ips = reply->request().rawHeader("Proof-IP-Addresses");
+    auto subject = QObject::tr("Network is slow!");
+    auto text = QStringLiteral("%1(%2)[%3]: %4, timeout: %5 msecs")
+                    .arg(slowNetworkAppId, ips, QDateTime::currentDateTimeUtc().toString(), reply->url().toString(),
+                         QString::number(timeout));
```

Comment:
Subj: Slow network access to HOST_NAME
Body:
Application: PRETTIFIED_APP_NAME (APP_ID)
OS: OS_NAME
IP: IPS_LIST_SEPARATED_WITH_SEMICOLON
Time: CURRENT_TIME_IN_WITH_TIMEZONE
URL: FULL_URL_WITH_QUERY


HOST_NAME should be hust a host name, no scheme, path or query included. FULL_URL should include all of them, but don't send request body.
I'm not sure about time though, if it is should be in local time or in utc. I would vote for UTC probably (and change it in email notifier to UTC too).